ansible-dk-cli/ansible-dk

```diff
@@ -105,15 +136,54 @@ def verify():
 def generate():
     pass
 
-@generate.command(name='playbook')
-@click.option('--name', required=True, help='Name of playbook to generate')
-def generate_playbook(name):
-    print("Generating playbook: ", name)
+def parse_extra_vars(context, param, e):
+    extra_vars = {}
+    for var in e:
+        try:
+            key, value = var.split('=')
+        except ValueError:
+            raise click.BadParameter('must be in the format var=value')
+        extra_vars[key] = value
+    return extra_vars
+
+def generate_method(thing):
+    @generate.command(name=thing)
+    @click.option('--debug', is_flag=True, help="Enable debug mode")
+    @click.option('--generator', '-g', default=get_config()['generator'], help='Path to custom generator repo')
+    @click.option('-e', 'extra_vars', multiple=True, callback=parse_extra_vars, help='extra vars to pass to ansible, var=value')
+    @click.argument('name')
+    def generated_method(debug, generator, name, extra_vars):
+        extra_vars[thing+'_name'] = name
+        playbook_path = "%s/%s.yml" % (generator, thing)
+        if debug:
+            click.echo('generating %s named %s using %s' % (thing, name, playbook_path))
+        run_ansible(playbook_path, extra_vars)
+    generated_method.__name__ = 'generate_%s' % thing
+
+generate_things = ['repo', 'role', 'filter', 'inventory', 'config', 'playbook']
+for thing in generate_things:
+    generate_method(thing)
 
-@generate.command(name='role')
-@click.option('--name', required=True, help='Name of role to generate')
-def generate_role(name):
-    print("Generating role: ", name)
+def run_ansible(playbook, vars=None):
+    if not os.path.exists(playbook):
+        print 'Error - specified role generator playbook does not exist'
+        sys.exit()
+    Options = namedtuple('Options', ['connection', 'module_path', 'forks', 'become', 'become_method', 'become_user', 'check', 'listhosts', 'listtasks', 'listtags', 'syntax', 'remote_user', 'private_key_file', 'ssh_common_args', 'sftp_extra_args', 'scp_extra_args', 'ssh_extra_args', 'verbosity'])
+    options = Options(connection='local', module_path=None, forks=100, become=None, become_method=None, become_user=None, check=False, listhosts=False, listtasks=False, listtags=False, syntax=False, remote_user=None, private_key_file=None, ssh_common_args=None, sftp_extra_args=None, scp_extra_args=None, ssh_extra_args=None, verbosity=None)
+    loader = DataLoader()
+    variable_manager = VariableManager()
+    inventory = Inventory(loader=loader, variable_manager=variable_manager, host_list='localhost')
+    variable_manager.set_inventory(inventory)
+    if vars:
```

Comment:
We will ALWAYS have vars to pass into the run.  We need to determine what they are, but a good start would be:

```
adk_generator_command        # 'role' in ansible-dk generate role pony
adk_generator_target_path     # ./pony if we're in repo mode, . otherwise 
adk_generator_target_name   # 'pony' in ansible-dk generate role pony 
adk_generator_source             # /opt/ansible-dk/generators/default or whatever              
```

The ansible-dk CLI will maintain a list of 'context vars' that it passes to the ansible run.  We must document these in the README or somewhere else.